### DISCUSSION TITLE
First issues on new image

### DISCUSSION THREAD

**Discussion by builderjer** [OpenVoiceOS/ovos-buildroot#106] [Q&A]
This is the new image, Rpi3b.
Everything booted, the WiFi setup worked well...after I found a Pi that the WiFi worked.
It even talked to me telling me that I need to set it up!!  That was a first on any of these that I have tried.
Setup went well, I setup WiFi on my phone, selected "Local Backend", and "Mimic2" as tts, because that is what the setup voice was and that is the first that I have heard a voice at all.
After the reboot, things looked good, but no voice, and no mic.  I have not looked into anything that I am stating here, just what happened as I started things.
I decided I would mess with the audio later, as that has been an issue with everything...always seems to be with anything linux.
So I dug into `~/.config/mycroft/mycroft.conf`.
I changed my location, because although the geolocate works with the IP, it is still 60 miles away from me. 
I performed `sudo systemctl restart mycroft` 
When everything started back, It showed that I needed to choose a backend again.  I went through this once before, but don't remember what I was testing.  I am assuming that this is something to do with [skill-ovos-setup](https://github.com/OpenVoiceOS/skill-ovos-setup)?
When I look at `~/.config/mycroft/mycroft.cong` again, all that was there was a list of blacklisted skills.  Also, there is something wrong there.  There is a whole bunch of duplicate entries there.
```
{
  "skills": {
    "blacklisted_skills": [
      "mycroft-pairing.mycroftai",
      "mycroft-volume.mycroftai",
      "mycroft-version-checker.mycroftai",
      "skill-wifi-connect.mycroftai",
      "mycroft-wiki.mycroftai",
      "mycroft-pairing.mycroftai",
      "mycroft-volume.mycroftai",
      "mycroft-version-checker.mycroftai",
      "mycroft-pairing.mycroftai",
      "mycroft-volume.mycroftai",
      "mycroft-version-checker.mycroftai",
      "skill-wifi-connect.mycroftai",
      "mycroft-wiki.mycroftai",
      "skill-repeat-recent",
      "mycroft-pairing.mycroftai",
      "mycroft-volume.mycroftai",
      "mycroft-version-checker.mycroftai",
      "mycroft-pairing.mycroftai",
      "mycroft-volume.mycroftai",
      "mycroft-version-checker.mycroftai",
      "skill-wifi-connect.mycroftai",
      "mycroft-wiki.mycroftai",
      "skill-repeat-recent",
      "mycroft-speak.mycroftai"
  ]
}
```
Nothing with the "tts" or "server" or anything

This is where I'm at now, going to blacklist the setup skill, and redo my mycroft.conf file and go from there.

These are logs since first boot
[logs.zip](https://github.com/OpenVoiceOS/OpenVoiceOS/files/8240789/logs.zip)
Will post more later

**Comment by j1nx** [comment]
Thanks for this! I converted this one to a discussion thread, so we can discuss the different aspect further and from that create one liner bug reports/issue that we can attach to the "fix for release" list.

Will go through the different aspect into detail later today.

**Comment by builderjer** [comment]
Next test:

Rpi4 8G
HDMI screen
Blue Snowball mic
Generic USB speaker
32G USB boot
Hard wired ethernet - no WiFi
Fresh image acquired from [here](https://drive.google.com/file/d/1Ez3Qa5P-fEA2PGunCdfTqawHy_62GpH3/view?usp=sharing)

First boot is awesome.  Boots quick, voice setup starts, Volume is LOUD!!  It didn't seem to be like that on the Rpi3 image.
Asks to choose a backend...Mimic2 voice
  - Local Backend
Ask to choose STT...Mimic2 voice
  - Google STT
No more voice!!!

I finish the setup,

Ask to choose TTS...NO VOICE
  - Larynx (online female)
    - I want this to work as I have a Larynx server running with a voice I like

The rest seems to work just fine, reboot went smooth, loaded the homescreen well, but NO SOUND and NO MIC.
I know this was discussed in chat, but here it is for a record.

**Comment by j1nx** [comment]
Yeah, indeed setting the volume to 50% on first boot is a known re-introduced bug. I will vcreate a ticket for it.

Sound issues: Hope to get to that this evening as it is the biggest showstopper bug at the moment. Thanks for testing! Trying to get another test image in your hands soon.

**Reply by builderjer** [reply]
Id love to build one, but it keeps getting stuck.  If you want to help me get past that point, I will give details in the issues

**Reply by j1nx** [reply]
Sure, but there are still a couple of loose end with some buildroot packages that I created which not yet really follow the proper coding style. There is another issue about that stuff already issued, but need to wait a little bit to go for a clean compile run so I know what I did to get past through it.

Focus for now is to get an image out, after that I will make time to fix the underlying build system.

**Comment by builderjer** [comment]
Does this image use HolmesV and not ovos-core?

**Reply by JarbasAl** [reply]
no

**Reply by builderjer** [reply]
Strange I found it installed. Maybe a unseen dependency somewhere?

**Reply by j1nx** [reply]
Probably a left over that I forgot to delete.

**Reply by j1nx** [reply]
Fixed in the next release. Forgot to delete the folder within the target image.

**Comment by builderjer** [comment]
Tracking down some TTS issues. It seems that the phrases are generated and put into the /tmp/TTS directory, but not read after that. The directory ~/.local/share/Mycroft/<TTS engine>/en-us/ is totally empty

**Reply by j1nx** [reply]
That second directory is more a location for a preloaded cache. Caches should be emptied on reboot and not write to SD as that will most likely fill up the card and perhaps even wreck it over a longer period of time.
